src/node_report.cc

```diff
@@ -0,0 +1,346 @@
+  fprintf(fd,"Process ID: %d\n", pid);
+
+  // Print Node.js and deps component versions
+  fprintf(fd,"\nNode.js version: %s\n", NODE_VERSION);
+  fprintf(fd,"(v8: %s, libuv: %s, zlib: %s, ares: %s)\n", V8::GetVersion(), uv_version_string(), ZLIB_VERSION, ARES_VERSION_STR);
+
+  // Print summary Javascript stack trace
+  fprintf(fd,"\n================================================================================\n");
+  fprintf(fd,"==== Javascript stack trace ====================================================\n\n");
+
+  switch(event) {
+  case kException:
+    // Print Javascript stack trace as stored in the V8 exception Message object. We need V8 to
+    // have saved the stack trace, see isolate->SetCaptureStackTraceForUncaughtExceptions().
+    if (!message_handle.IsEmpty()) {
+      PrintStackFromMessage(fd, isolate, message_handle);
+    }
+    break;
+  case kFatalError:
+    // Print Javascript stack using the Message::PrintCurrentStackTrace() API
+    Message::PrintCurrentStackTrace(isolate, fd);
+    break;
+  case kSignal_JS:
+  case kSignal_UV:
+    // Print Javascript stack using the StackTrace::StackTrace() and GetStackSample() APIs
+    PrintStackFromStackTrace(fd, isolate, event);
+    break;
+  } // end switch(event)
+
+  // Print V8 Heap and Garbage Collector information
+  PrintGCStatistics(fd, isolate);
+  
+  // Print operating system information
+  PrintSystemInformation(fd, isolate);
+
+  // TODO: Obtain and print libuv information, e.g. uv_print_all_handles()
+  // TODO: Print native stack
+  // TODO: Obtain and print CPU time information e.g. getrusage(), and/or the new V8 CPU time APIs
+
+  fprintf(fd,"\n================================================================================\n");
+  fclose(fd);
+  fprintf(stderr,"Node.js error report completed\n");
+}
+
+/*
+ * Function to print stack trace contained in supplied V8 Message object.
+ */
+static void PrintStackFromMessage(FILE* fd, Isolate* isolate, Local<Message> message_handle) {
+
+  Local<String> message_string = message_handle->Get();
+  node::Utf8Value message_utf8(isolate, message_string);
+  Local<StackTrace>stack = message_handle->GetStackTrace();
+
+  if (stack.IsEmpty()) {
+    fprintf(fd,"\nNo stack trace available from V8 Message: %s\n", *message_utf8);
+    return;
+  }
+  fprintf(fd,"V8 Message: %s\n", *message_utf8);
+  for (int i = 0; i < stack->GetFrameCount(); i++) {
+	  PrintStackFrame(fd, isolate, stack->GetFrame(i), i);
+  }
+}
+
+/*
+ * Function to print stack trace  the StackTrace::StackTrace() and GetStackSample() APIs
+ */
+static void PrintStackFromStackTrace(FILE* fd, Isolate* isolate, DumpEvent event) {
+  v8::RegisterState state;
+  v8::SampleInfo info;
+  void* sample[255];
+
+  // Initialise the register state
+  state.pc = NULL;
+  state.fp = &state;
+  state.sp = &state;
+
+  isolate->GetStackSample(state, sample, 255, &info);
+  fprintf(fd,"Javascript VM state: %s\n\n", v8_states[info.vm_state]);
```

Comment:
Good idea, in case more state values are added. CHECK routes to V8_Fatal(), alternative for this kind of thing is to make the report code resilient, print 'unknown' and carry on.